dagshub/data_engine/model/datasource.py

```diff
@@ -352,6 +374,43 @@ def send_datapoints_to_annotation(self, datapoints: Union[List[Datapoint], List[
             webbrowser.open_new_tab(link)
         return link
 
+    def wait_until_ready(self, max_wait_time=300, fail_on_timeout=True):
+        """
+       Blocks until the datasource preprocessing is complete
```

Comment:
Blocks what? This needs a more detailed explanation. Why would someone use this?